### PR TITLE
Extract API classes into their own artifact

### PR DESCRIPTION
**Pull request by kenfinnigan**
- Makes it easier to identify possible spec proposals in the future
 - Deprecated Emitter in annotations package, and created copy in main package as replacement

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,21 +56,26 @@
...
-    <cloudevent.version>0.2.1</cloudevent.version>
-    <axle.version>0.0.10</axle.version>
 
     <asciidoctorj.version>1.5.8.1</asciidoctorj.version>
 
     <sonar.coverage.jacoco.xmlReportPaths>target/site/jacoco/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
   </properties>
 
   <modules>
+    <module>reactive-messaging-api</module>
```
Comment (cescoffier) [inline pom.xml]: should it be named smallrye-reactive-messaging-api?

File: smallrye-reactive-messaging-provider/pom.xml [hunk 2]
```diff
@@ -13,13 +13,18 @@
...
-      <artifactId>jandex</artifactId>
+      <groupId>io.smallrye.reactive</groupId>
+      <artifactId>smallrye-reactive-messaging-api</artifactId>
     </dependency>
 
     <dependency>
-      <groupId>javax.annotation</groupId>
-      <artifactId>javax.annotation-api</artifactId>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/pom.xml]: Any reason for this dependency?

**Comment by kenfinnigan** [comment]
probably, let me check docs

File: pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply pom.xml]: the artifact is named that, I just cheated on the directory name to have it at the top.

Can add "smallrye" for consistency if desired

File: smallrye-reactive-messaging-provider/pom.xml (hunk 2 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-provider/pom.xml]: `MediatorConfigurationSupport` uses `ClassUtils` from it.

Comment (cescoffier) [reply smallrye-reactive-messaging-provider/pom.xml]: We should consider removing it at some point. Depending on the lib for a single class, does not make a lot of sense.

**Comment by kenfinnigan** [comment]
Totally agree. I can look at that this week

**Comment by cescoffier** [comment]
Can you resolve the conflict?

**Pull request merged**